### PR TITLE
Update zeitschrift-fur-kunstgeschichte.csl

### PR DESCRIPTION
**Pull request by blitzgneisserin**
I added the url to online-publications (or publications that were also published online eventually);
I added the university and "Diss." to theses;
I corrected the sorting order of the bibliography (in the previous version edited books sometimes landed at the end): I think the bibliography should be sorted by author/editor and year with the exception of exhibition catalogues which should be at the end of the bibliography and sorted by year (and title); I have pasted some code form "Geistes- und Kulturwissenschaften (Heilmann)", it appears to work correctly but I am not sure if it is the best solution.

Please note that I will not be able to make any corrections for a few days from tomorrow because I will not have a computer.

### CODE REVIEW

File: zeitschrift-fur-kunstgeschichte.csl [hunk 1]
```diff
@@ -204,12 +237,25 @@
...
+          </group>
+        </if>
+      </choose>
     </layout>
   </citation>
   <bibliography et-al-min="3" et-al-use-first="1" subsequent-author-substitute="&#8212;&#8212;&#8212;" entry-spacing="0" hanging-indent="true">
     <sort>
-      <key macro="author"/>
-      <key variable="issued"/>
+      <key macro="contributors-long" names-min="3" names-use-first="3"/>
```
Comment (zuphilip) [inline zeitschrift-fur-kunstgeschichte.csl]: Why did you change the et-al rule here? That looks wrong for me.

I have seen in some examples:
 * 1 author: `author1`
 * 2 authors: `author1 and author2`
 * 3+ authors: `author1 et al.`

Comment (blitzgneisserin) [inline zeitschrift-fur-kunstgeschichte.csl]: This is the code that I copied from "Geistes- und Kulturwissenschaften (Heilman)", I did not change the rule intentionally. Is it such a big mistake?
Well I guess names-min="3" and names-use-first="3" are unnecessary.

Comment (zuphilip) [inline zeitschrift-fur-kunstgeschichte.csl]: The copied code would produce something like `author1, author2, author3 et al.` which is wrong here for this journal.

Since `et al.` is standing for "and others" it is normally used only to abbreviate more than one person. Thus, there is normally at least a difference of 2 between these two values.

Yes, please delete those two attributes again. We already have the correct values in the bibliography node above.

**Comment by adam3smith** [comment]
Sorry, but could we step back and describe exactly how we want this sorted and what the current problem is? It looks to me like we'd want fairly standard sorting behavior and I don't see any reason to just copy over entire macros from another style just for sorting. This should be possible by using existing macros, no?

**Comment by blitzgneisserin** [comment]
It is a fact that edited books are landing at the end of the bibliography together with the exhibition catalogues, why exactly I am not sure. Because of the special rule for the exhibition catalogues (they do not begin with the editors but with the title, thats why zuphilip chose wisely the type entry-encyoclopedia for them) it is clear though that a normal author-or-editor-macro does not work here. The macros that I found in the mentioned style do make a lot of sense for me. Unfortunately I cannot write detailed explanations here because I only have a rather small tablet. One more thingthough: the journal does not require a bibliography which is why it does not  recommend anything for the sorting of the bibliography. I think however that exhibition catalogues should be at the end of the bibliography etc see above.

**Comment by zuphilip** [comment]
Here is an example for the ordering in the bibliography at the moment:

```
T Accadia et al., Virgo: a laser interferometer to detect gravitational waves, in: Journal of Instrumentation 7/03, 2012, P03012–P03012.
Jorge Luis Borges, Selected non-fictions, ed. by Eliot Weinberger, trans. by Esther Allen et al., New York 1999.
John L. Campbell and Ove K. Pedersen, The varieties of capitalism and hybrid success, in: Comparative Political Studies 40/3, 2007, 307–332.
Nigel Dunnett and Noël Kingsbury, Planting green roofs and living walls, Portland, OR 22008.
Walter Isaacson, Steve Jobs, New York, NY 2011.
Isabela Mares, Firms and the welfare state: When, why, and how does social policy matter to employers?, in: Peter A Hall and David Soskice (Hg.), Varieties of capitalism. The institutional foundations of comparative advantage, New York 2001, 184–213.
Maurie Dee McInnis and Louis P. Nelson, Shaping the body politic: Art and political formation in early america, Charlottesville, VA 2011.
Bob Hancké et al. (Hg.), Beyond varieties of capitalism: conflict, contradiction, and complementarities in the European economy, Oxford and New York 2007.
CSL search by example, in: Citation Style Editor.
```
I don't know what a good way to order this actually...

**Comment by blitzgneisserin** [comment]
Well?
I see one more improvement: uniting the two macros "contributors-long" and "yearandtitle". Would that work?
I think that my suggestions for ordering the bibliography are not so bad. Usually exhibition and collection catalogues have separate subchapters within the bibliography, but I think that is not possible with Zotero respectively this citation style.

**Comment by adam3smith** [comment]
I guess my first concern would be that I don't think this makes a lot of sense as a bibliography that's going to be ordered alphabetically in the first place: if you want to do that, you want to have at least the first author as lastname, initial (name-as-sort-order="first")

**Comment by blitzgneisserin** [comment]
So is it better to do nothing, leave it as it is? Or do we leave the bibliography?
Obviously we are used to different norms and habits...

**Comment by blitzgneisserin** [comment]
I have just seen an interesting system in "Abstract Art" by Anna Moszynska: maybe we could make the first author/editor as lastname, firstname and put the exhibition catalogues (which do not begin with the editor's name) inbetween the entries with author/editor, depending on the first letter/word of the title (i.e. order them alphabetically together with the other entries)?

**Comment by adam3smith** [comment]
yes, that sounds good to me. @zuphilip -- you have worked more with the style, could you look at implementation?
Ideally I'd like to see the author macro consolidated to include the catalog exception for editors. Then we could create author-bibliography as a new macro and make it the same except for `name-as-sort-order`.  We could then use that for sorting. 
I'd then also create a year-date macro instead of all the individual variable calls to `issued` and use that for sort. Is that feasible?

**Comment by blitzgneisserin** [comment]
I have tried something. I think the result is quite close but I have not achieved it exactly like you suggested.
Here is the modified code (I have not shared it here on github since I am not sure if it is ok):
http://simona.bplaced.net/zotero/zeitschrift-fur-kunstgeschichte26.csl
And this is the result:

Der Meister von Flémalle und Rogier van der Weyden (Ausst.-Kat. Frankfurt und Berlin), hg. von Stephan Kemperdick und Jochen Sander, Ostfildern 2008.
Elger, Dietmar und Uta Grosenick (Hg.), Abstrakte Kunst, Köln 2012.
Hornuff, Daniel, Bildwissenschaft im Widerstreit: Belting, Boehm, Bredekamp, Burda, Paderborn 2012.
Moszynska, Anna, Abstract Art, London ; New York, N.Y 1990.
Simon, Anna, Studien zu Hugo van der Goes, Diss., Staatliche Akademie der Bildenden Künste Stuttgart, 2015, http://archiv.ub.uni-heidelberg.de/artdok/volltexte/2015/3452 (letzter Zugriff am 27.1.2016).
———, [Rezension von:] Das altniederländische Stifterbild : Emotionsstrategien des Sehens und der Selbsterkenntnis / Johanna Scheel. - 1. Aufl. . - Berlin : Mann, 2014. - 548 S. : Ill. . - ISBN: 978-3-7861-2695-9. - (Neue Frankfurter Forschungen zur Kunst ; 14), in: Journal für Kunstgeschichte 18, 2014, 226–231.
———, [Rezension von:] The road to Van Eyck : [catalogue published to coincide with the exhibition of the same name in Museum Boijmans Van Beuningen, Rotterdam, 13 October 2012 - 10 February 2013] / Stephan Kemperdick ; Friso Lammertse. - Rotterdam : Museum Boijmans van Beuningen, 2012. - 342 S. : zahlr. Ill. . - ISBN: 978-90-6918-262-9, in: Journal für Kunstgeschichte 17/4, 2013, 258–268.
The road to Van Eyck (catalogue published to coincide with the exhibition of the same name in Museum Boijmans Van Beuningen, Rotterdam, 13 October 2012 - 10 February 2013), hg. von Stephan Kemperdick und Friso Lammertse, Rotterdam 2012.
Vroege Hollanders : schilderkunst van de late Middeleeuwen (gepubliceerd ter gelegenheid van de tentoonstelling „Vroege Hollanders ...“ die van 16 februari tot en met 25 mei 2008 plaatsvindt in Museum Boijmans Van Beuningen in Rotterdam), hg. von Friso Lammertse und Jeroen Giltaij, Rotterdam 2008.

As you can see there is one new macro "sort-bibliography", I have deleted the macros "contributors-short" and "yearandtitle".

**Comment by zuphilip** [comment]
Sorry, I am currently swamped with other stuff. I have read the plan for an sorting and that sounds reasonable. I will not be able to look at the technical implementation any time soon.

**Comment by adam3smith** [comment]
It's been a hectic month for me. I have the weekend budgeted to get through lots of things left behind, so will take this then. Thank you for your patience.

**Comment by blitzgneisserin** [comment]
no problem

**Comment by adam3smith** [comment]
sorry for the delay. I have now looked at that and have a revised version ready to put up, but in the course I have discovered what I believe to be a bug in the citation processor (described here: https://forums.zotero.org/discussion/57100/citeproc-bug-substitute-doesnt-inherit-label-with-conditional/ ) and I want to wait to hear back from that before putting my version of the style up.

**Comment by blitzgneisserin** [comment]
so?

**Comment by adam3smith** [comment]
see the message there, I'm holding off until I can get the fix confirmed.

**Comment by rmzelle** [comment]
Thanks, @blitzgneisserin!
